### PR TITLE
Testing payment using standard stripe UI components

### PR DESCRIPTION
**Pull request by prasanthu**
Add support for payment using Stripe Standard UI Components

This is currently used only for testing. To enable this feature add "-DSTRIPE_PAYMENT" to "Other Swift Flags" in Build Settings for the target being tested.

### CODE REVIEW

File: ostelco-ios-client/AppDelegate.swift [hunk 1]
```diff
@@ -23,7 +23,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
         STPPaymentConfiguration.shared().publishableKey = Environment().configuration(.StripePublishableKey)
+        #if STRIPE_PAYMENT
+        debugPrint("Stripe Payment enabled")
+        #else
```
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/AppDelegate.swift]: That is how Xcode intends `#if` blocks. The Xcode `Re-Indent` will revert this back to this style.

[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/AppDelegate.swift]: Is there a style guide I can refer to ?

File: ostelco-core/Network/LoggedInAPI.swift [hunk 2]
```diff
@@ -105,7 +105,15 @@ open class LoggedInAPI: BasicNetwork {
                 debugPrint("Delete customer response: \(String(describing: dataString))")
             }
     }
-    
+
+    /// - Returns: A Promise which when fulfilled will contain the Stripe Ephemeral Key
+    public func stripeEphemeralKey(apiVersion: String) -> Promise<[String: AnyObject]?> {
```
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-core/Network/LoggedInAPI.swift]: a) That is how stripe expects the `EphemeralKey`

[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-core/Network/LoggedInAPI.swift]: b) Will do

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift [hunk 3]
```diff
@@ -61,3 +77,95 @@ extension ApplePayViewController: PKPaymentAuthorizationViewControllerDelegate {
         handlePaymentFinished(controller)
     }
 }
+
+#if STRIPE_PAYMENT
```
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: Didn't want this to be in the released binary...

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift [hunk 4]
```diff
@@ -61,3 +77,95 @@ extension ApplePayViewController: PKPaymentAuthorizationViewControllerDelegate {
...
+        let paymentOptionsViewController = STPPaymentOptionsViewController(paymentContext: paymentContext)
+        let navigationController = UINavigationController(rootViewController: paymentOptionsViewController)
+        present(navigationController, animated: true)
+    }
+    // Do actual payment using  Stripe.
+    // Payment is done using the default card you have selected using the showPaymentOptions API
+    // If no card is present, this will show the payment options UI
+    func startStripePay(product: Product) {
+        purchasingProduct = product
+        paymentContext.paymentAmount = Int(truncating: NSDecimalNumber(decimal: purchasingProduct!.amount))
```
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: done

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift [hunk 5]
```diff
@@ -61,3 +77,95 @@ extension ApplePayViewController: PKPaymentAuthorizationViewControllerDelegate {
...
+        paymentContext.paymentCurrency = purchasingProduct!.currency
+        paymentContext.requestPayment()
+    }
+    // Stripe is ready with a payment source, call Prime API to purchase the product
```
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: done

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift [hunk 6]
```diff
@@ -61,3 +77,95 @@ extension ApplePayViewController: PKPaymentAuthorizationViewControllerDelegate {
...
+        @unknown default:
+            debugPrint("Payment finished with unknown status \(status)")
+        }
+    }
+
+}
+
+// This class provides the ephemeral key for Stripe payments
+// Called automatically by Stripe through the STPCustomerContext object
+class StripeKeyProvider: NSObject, STPCustomerEphemeralKeyProvider {
```
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: done, I was blindly following their example.

File: ostelco-ios-client/AppDelegate.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (designatednerd) [reply ostelco-ios-client/AppDelegate.swift]: OMFG. I need to file a radar for that then, that's absolute nonsense. There isn't a particular style guide which requires this but I've generally found it to be WAY easier to read (and to also make places you're using these `#if` directives jump out at you)

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (designatednerd) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: Sort of makes sense, but at a certain point, why wouldn't we want this in the binary if we're actually going to use it?

File: ostelco-ios-client/AppDelegate.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/AppDelegate.swift]: Do you want it like normal `if` indentation or should it be like `#if` always starts at column 1

[the remark this answers is not in the document]
Comment (designatednerd) [reply ostelco-ios-client/AppDelegate.swift]: i think normal if? [Here's an example from my open PR](https://github.com/ostelco/ostelco-ios-client/pull/192/files#diff-5dd63a00b3f041b84d7215f56d36e4a9R20)

File: ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift]: When we decide to add other methods of payments, let's drop the #if altogether.

**Pull request merged**